src/e2e/claim.rs

```diff
@@ -0,0 +1,302 @@
+        assert_eq!(amount_y, U256::from(1000));
+
+        let pool_before = invariant
+            .get_pool(pool_key.token_x, pool_key.token_y, fee_tier)
+            .unwrap();
+
+        let slippage = SqrtPrice::new(U128::from(MIN_SQRT_PRICE));
+        let swap_amount = TokenAmount::new(amount);
+        invariant
+            .swap(pool_key, true, swap_amount, true, slippage)
+            .unwrap();
+
+        let pool_after = invariant
+            .get_pool(pool_key.token_x, pool_key.token_y, fee_tier)
+            .unwrap();
+
+        let expected_tick = -20;
+
+        assert_eq!(pool_after.liquidity, pool_before.liquidity);
+        assert_eq!(pool_after.current_tick_index, expected_tick);
+        assert_ne!(pool_after.sqrt_price, pool_before.sqrt_price);
+
+        let amount_x = token_x.balance_of(&caller);
+        let amount_y = token_y.balance_of(&caller);
+        assert_eq!(amount_x, U256::from(0));
+        assert_eq!(amount_y, U256::from(993));
+
+        let amount_x = token_x.balance_of(invariant.address());
+        let amount_y = token_y.balance_of(invariant.address());
+        assert_eq!(amount_x, U256::from(1500));
+        assert_eq!(amount_y, U256::from(7));
+
+        assert_eq!(
+            pool_after.fee_growth_global_x,
+            FeeGrowth::new(U128::from(50000000000000000000000u128))
+        );
+        assert_eq!(
+            pool_after.fee_growth_global_y,
+            FeeGrowth::new(U128::from(0))
+        );
+
+        assert_eq!(
+            pool_after.fee_protocol_token_x,
+            TokenAmount::new(U256::from(1))
+        );
+        assert_eq!(
+            pool_after.fee_protocol_token_y,
+            TokenAmount::new(U256::from(0))
+        );
+    }
+    // Claim fee
+    {
+        let position_owner = test_env::get_account(0);
+        test_env::set_caller(position_owner);
+        let user_amount_before_claim = token_x.balance_of(&position_owner);
+        let dex_amount_before_claim = token_x.balance_of(invariant.address());
+
+        invariant.claim_fee(0).unwrap();
+
+        // Load states
+        let pool = invariant
+            .get_pool(*token_x.address(), *token_y.address(), fee_tier)
+            .unwrap();
+        let position = invariant.get_position(0).unwrap();
+        let user_amount_after_claim = token_x.balance_of(&position_owner);
+        let dex_amount_after_claim = token_x.balance_of(invariant.address());
+        let expected_tokens_claimed = U256::from(5);
+
+        assert_eq!(
+            user_amount_after_claim - expected_tokens_claimed,
+            user_amount_before_claim
+        );
+        assert_eq!(
+            dex_amount_after_claim + expected_tokens_claimed,
+            dex_amount_before_claim
+        );
+        assert_eq!(position.fee_growth_inside_x, pool.fee_growth_global_x);
+        assert_eq!(position.tokens_owed_x, TokenAmount::new(U256::from(0)));
+    }
+}
+
+#[test]
+fn test_claim_not_admin() {
```

Comment:
Maybe change the name to `..._not_owner`. By the name, I am wondering if this refers to the withdrawal fee from the pool by the admin or the withdrawal fee accumulated by the owner's position